docs/metasploit-framework.wiki/dev/Setting-Up-a-Metasploit-Development-Environment.md

```diff
@@ -129,27 +127,103 @@ Congratulations! You have now set up a development environment and the latest ve
 
 ## Optional: Set up the REST API and PostgreSQL database
 
-The following optional section describes how to manually install PostgreSQL and set up the Metasploit database.  Alternatively, use our Omnibus installer which handles this more reliably.
+Installing the REST API and PostgreSQL is optional, and can be done in two ways.
+Recommended is to use the Docker approach, and fairly simple to do once you have docker installed on your
+system, [Docker Desktop][docker-desktop] is recommended, but not mandatory.
+On Linux systems, simply having docker-cli is sufficient.
+
+### Docker Installation
+
+**Make sure, you have docker available on your system: [Docker Installation Guide][docker-installation]**
+
+**Note**: Depending on your environment, these commands might require `sudo`
+
+* Pull the Docker container: `docker pull postgres:14`
+* Start the container:
+
+```bash
+docker run --name "postgres" -p 5432:5432 -e POSTGRESS_PASSWORD="mysecretpassword" -d postgres:14
+```
+
+Wait till the container is fully running.
+We need to make some changes in the container to be able to connect to it remotely using the `psql` client.
+
+* Open a shell to the running container: `docker exec -it "postgres" bash`
+* Install an editor: `sudo apt update && sudo apt install -y nano`
+* Edit `/var/lib/postgresql/data/postgresql.conf` and uncomment: `# password_encryption = scram-sha-256	# scram-sha-256 or md5`
+* Exit the shell on the container
+* Ensure the postgres client is installed on your host system
+* connect to the running container (the password is `mysecretpassword`): `psql -h 127.0.0.1 -U postgres`
+* Run the following commands:
+
+```bash
+CREATE USER metasploit ENCRYPTED PASSWORD 'rapid7';
+ALTER USER metasploit WITH Superuser;
+ALTER USER metasploit WITH CREATEDB;
+```
+
+* Verify the user has been successfully created:
+
+```bash
+postgres-# \du
+                              
+ Role name  |                      List of roles Attributes                         
+------------+------------------------------------------------------------
+ metasploit | Superuser, Create DB
+ postgres   | Superuser, Create role, Create DB, Replication, Bypass RLS
+```
```

Comment:
Can we skip all these steps by just updating database.yml to use postgres:myseretpassword directly? I don't think adding this extra layer of user configuration adds anything here in terms of security if we treat the postgres instance as ephemeral

Reply:
I mean, technically yes, you could run everything under the `postgres` user and do it that way.
As long as the user knows what they are doing it will work.